_docs/concepts/encryption.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: guide
+title: StorageOS Docs - Encryption
+anchor: concepts
+module: concepts/encryption
+---
+
+# Encryption
+
+Encryption is enabled on a per volume basis using the
+`storageos.com/encryption` label (for more information see [Encrypted
+Volumes](/docs/operations/encrypted-volumes)).
+
+## Encryption Keys
+
+StorageOS encrypts volumes on disk using AES-256 in XTS-AES mode with 512 bit
+keys as recomended by [NIST](
+https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38e.pdf
+), with encryption keys being derived using [HKDF](
+https://eprint.iacr.org/2010/264.pdf ).
+
+Keys and Initialisation vectors are generated using the [crypto/rand](
+https://godoc.org/crypto/rand ) package.
+
+| Key                   | Size                 | Usage                                                     |
+| :-------------------- | :------------------- | :-------------------------------------------------------- |
+| Volume                | 512 bits random data | Used by XTS-AES to encrypt and decrypt disk blocks        |
+| User                  | 256 bits random data | Used to encrypt Volume keys                               |
+| Initialisation Vector | 256 bits random data | Used as 'salt' when encrypting volume keys with user keys |
+
+The components required to derive the encryption key are stored in a Kubernetes
+secret. By default these secrets are stored in the namespace that StorageOS is
+installed into. As Kubernetes secrets are only base64 encoded, it is recommend
+to [encrypt secrets at
+rest](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/), this
+can be done using a
+[KMS](https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/) such
+as [HashiCorp Vault](https://www.vaultproject.io/).
+
+## Namespace Specific Keys
+
+Each namespace has its own user key that is created when the namespace is
+initialized. The namespace keys are stored as Kubernetes secrets named
+`ns-key.{Namespace Name}`. The user key secrets are created in whatever
+namespace StorageOS is installed into.
+
+In the example below there are two `ns-key` secrets in the `storageos`
+namespace because a StorageOS volume has been provisioned in the `default` and
+`mongo` namespaces. A `ns-key` is created for a namespace regardless of whether
+an encrypted volume exists in the namespace or not.
+
+```bash
+$ kubectl get secrets -n storageos | grep ns-key
+ns-key.default                                 Opaque                                1      4d
+ns-key.mongo                                   Opaque                                1      5h
+```
+
+## Encypted Volume life cycle
+
+A volume is encrypted with a volume key that is a randomly generated 512 bit
+key. Rather than storing the volume key, StorageOS stores an encrypted version
+of the volume key, called the volume user key, which is generated by encrypting
+the volume key with a 256 bit user key and 256 bit initialization vector. Each
```

Comment:
Seems to me that user key and namespace key mean the same thing here. Why don't we just say that 'which is generated by encrypting the volume key with a 256 bit namespace key and 256 bit initialisation vector' ?

Reply:
(initalialization - sorry, force of habit)